usbutils.pc.in

```diff
@@ -15,4 +15,4 @@ Requires: libusb-1.0 >= 1.0.14  libudev >= 196
 Conflicts:
 Libs: -L${libdir}
 Libs.private: @LIBUSB_LIBS@ @UDEV_LIBS@
-Cflags: @CFLAGS@ @LIBUSB_CFLAGS@ @UDEV_CFLAGS@
```

Comment:
Don't we need the libusb and udev cflags, why are you removing them?

Also, I do not see a tmpdir here, where is that coming from?

Reply:
Hi @gregkh 

According to https://people.freedesktop.org/~dbn/pkg-config-guide.html:

"Cflags: The compiler flags specific to this package and any required libraries that don't support pkg-config. If the required libraries support pkg-config, they should be added to Requires or Requires.private."

Both libusb and udev support pkg-config. They are already present in the Requires entry.

That's the reason for the removal.

I can add this information to the commit log.

About the tmpdir error, this comes from OpenEmbedded. Maybe @kraj can explain it better.

Thanks

Reply:
> Hi @gregkh
> 
> According to https://people.freedesktop.org/~dbn/pkg-config-guide.html:
> 
> "Cflags: The compiler flags specific to this package and any required libraries that don't support pkg-config. If the required libraries support pkg-config, they should be added to Requires or Requires.private."
> 
> Both libusb and udev support pkg-config. They are already present in the Requires entry.
> 
> That's the reason for the removal.
> 
> I can add this information to the commit log.
> 
> About the tmpdir error, this comes from OpenEmbedded. Maybe @kraj can explain it better.
> 
> Thanks

The problem with cross building is that these flags expands to absolute paths upon final install of .pc file
but I think that issue is specific to cross-build systems as native builds will use /usr for sysroot and all falls in place. 

@fabioestevam whats the content of LIBUSB_CFLAGS UDEV_CFLAGS ? maybe we can post-process these variables to edit out the absolute path.

Reply:
Hi @kraj 

Inspecting the generated usbutilis.pc, the Cflags content is:

```
Cflags: -O2 -pipe -g -feliminate-unused-debug-types -fcanon-prefix-map -fmacro-prefix-map=/work/festevam/oe/poky/build/tmp/work/core2-64-poky-linux/usbutils/017/git=/usr/src/debug/usbutils/017 -fdebug-prefix-map=/work/festevam/oe/poky/build/tmp/work/core2-64-poky-linux/usbutils/017/git=/usr/src/debug/usbutils/017 -fmacro-prefix-map=/work/festevam/oe/poky/build/tmp/work/core2-64-poky-linux/usbutils/017/build=/usr/src/debug/usbutils/017 -fdebug-prefix-map=/work/festevam/oe/poky/build/tmp/work/core2-64-poky-linux/usbutils/017/build=/usr/src/debug/usbutils/017 -fdebug-prefix-map=/work/festevam/oe/poky/build/tmp/work/core2-64-poky-linux/usbutils/017/recipe-sysroot= -fmacro-prefix-map=/work/festevam/oe/poky/build/tmp/work/core2-64-poky-linux/usbutils/017/recipe-sysroot= -fdebug-prefix-map=/work/festevam/oe/poky/build/tmp/work/core2-64-poky-linux/usbutils/017/recipe-sysroot-native= -I/work/festevam/oe/poky/build/tmp/work/core2-64-poky-linux/usbutils/017/recipe-sysroot/usr/include/libusb-1.0
```

Is it valid to pass Cflags: @CFLAGS@ @LIBUSB_CFLAGS@ @UDEV_CFLAGS@ ?

According to the Cflags explanation in  https://people.freedesktop.org/~dbn/pkg-config-guide.html, we should not pass @LIBUSB_CFLAGS@ @UDEV_CFLAGS@ .

Thanks

Reply:
My two cents:

There's no need to pass `@CFLAGS@` as that's the flags used to _compile_ usbutils, which are not relevant when compiling something _against_ libusb: just because libusb was built with `-O2` doesn't mean that flag should be passed to my build.  The failure case for us is because `CFLAGS` contains debug path mappings, so build paths leak into the target packages.

`@LIBUSB_CFLAGS` and `@UDEV_CFLAGS@` are not needed (also `LIBUSB_LIBS` and `UDEV_LIBS`) because those libraries are listed in the `Requires` field, so pkg-config will fetch the flags needed via those.

Reply:
Hi @rossburton 

Thanks for your feedback.

If I understood you correctly, we could remove the Cflags: entry then. Is this correct?

Or should we go with: Cflags: -I${includedir} as I did on this patch?

Thanks